server/services/matchChat.ts

```diff
@@ -0,0 +1,58 @@
+import { IMatchMessage } from '../interfaces';
+import { Message } from '../models';
+import { CustomError } from '../utils';
+
+const addMessageService = async ({
+  message,
+  senderId,
+  matchId,
+}: IMatchMessage) => {
+  const newMessage = await Message.create({
+    UserId: senderId,
+    MatchId: matchId,
+    message,
+  });
+
+  return newMessage;
+};
+
+const getMessageByIdService = async (messageId: number) => {
+  const message = await Message.findOne({ where: { id: messageId } });
+
+  return message;
+};
+
+const getAllMessagesService = async () => {
+  const messages = await Message.findAll();
+
```

Comment:
You must take in consederation the not happy reults in your services, like returning no messages, in this case you need to return message means that there is no any messages! 
and return this form of response from the service : 
```
//in failure case  
return {
      status: errorStatue[ex: 404],
      data: ''no messages !",
    };
    //in success case  
return {
      status: success Status [ex:  200],
      data: result,
    };

```  
   and handle it in the controller in this way : 
   
```
 const data = (await yourService(req));

  res.status(data.status).json(data);
```
  
  this way will help you also to send the right status and clear response for the frontEnd Side
  
 ### Note : 
  - take in consideration that make this in all services and controllers